### PR TITLE
Joe

### PR DESCRIPTION
**Pull request by fractalscape13**
This commit gets the chat screen to default to scroll to the bottom of the list, showing the newest message.  When a new message is added, it immediately scrolls to the bottom showing the new message.  Trying to figure out how to have the list immediately at the bottom instead of showing the scroll animation, but not having luck with that yet.

### CODE REVIEW

**Comment by fractalscape13** [comment]
Moved the text input field into the KeyboardAwareScrollView and updated the margin styling so the text input is visible on iOs and Android when the keyboard is being used

**Comment by fractalscape13** [comment]
Most recent commit adds differing colors and margins for sender/recipient messages.  Text input is visible when keyboard is visible, and messages default to showing the newest.  There is still an issue with scrolling: when you scroll up in the messages, the text input disappears, and you have to scroll down to see it.  I tried many options to remedy this and have not been fully successful yet.  Will continue to experiment

**Comment by fractalscape13** [comment]
This commit allows both Android and iOS to scroll without moving the TextInput, and when the keyboard is shown, the TextInput is visible.  Android was working without any additional attributes on the KeyboardAvoidingView, but iOS required adding behavior and keyboardVerticalOffset attributes to the parent element.  There is a ternary in the behavior attribute that checks the platform and adjusts accordingly.  The screen shots show the message list scrolled up with the text input still showing, as well as the keyboard showing with the text input visible and the normal view where the newest message is shown at the bottom.
Screen Shot 2020-06-10 at 11 28 00 AM
Screen Shot 2020-06-10 at 11 28 17 AM
Screen Shot 2020-06-10 at 11 28 41 AM

File: screens/messaging/ChatScreen.js [hunk 1]
```diff
@@ -121,101 +127,139 @@ function ChatScreen(props, { getMessages, postMessages, readMessage }) {
...
-                  ) : (
-                    // todo: filled in check circle when read, outline when not
-                    <Text></Text>
-                  )}
+    <KeyboardAvoidingView 
+      style={styles.container}
+      behavior={Platform.OS == 'ios' ? behavior : null}
+      keyboardVerticalOffset={180}>
+      <ScrollView 
+        onKeyboardWillShow={openKeyboard}
```
Comment (LeePedersen) [inline screens/messaging/ChatScreen.js:146]: I think you should be able to get rid of the openKeyboard and closeKeyboard functions, like by replacing line 146 with         onKeyboardWillShow={() => setBehavior("padding")}, which seems a bit DRYer to me. But let me know if there are issues with that

**Review by LeePedersen — COMMENTED** [review]
Looks great, I'm impressed. Quick note -  we're trying to use double quotes for strings for consistency (to be clear, I'm not asking you to change this, just for future reference)

**Pull request merged**